### PR TITLE
netlify adapter - add esbuild plugin to allow `node:` prefixed imports

### PR DESCRIPTION
**Pull request by theoephraim**
Addresses #285 

## Changes

Adds a simple esbuild plugin to allow `node:` prefixed imports in the netlify adapter's call to esbuild.

## Testing

Tests are currently not passing.

I started to play with it but it seems to do more with how the testing is set up and fixtures are loaded. I would have expected the middleware having node imports in it to not trip up at all, and only falter on the unprefixed node import, only when `edgeMiddleware: true` option is on.

Happy to help collaborate to fix it, but someone with more knowledge of the testing setup will be much better able to solve this.

I have personally verified that adding this code fixes my issues when building and deploying to netlify, and does not seem to break anything else.

## Docs

I think this change should match user's expectations moreso than the current behaviour.

### CODE REVIEW

**Review by alexanderniebuhr — COMMENTED** [review]
The change looks good to me, we need to figure out why tests are broken. I won't have time for that soon, but I posted this in our Discord to see if anyone else wants to jump in.

**Comment by alexanderniebuhr** [comment]
I will probably find time to look at the test soon. However wondering to get @ascorbic review of the general approach of allowing `node:` prefix. Should be safe, I guess :)

**Comment by ascorbic** [comment]
Yes, the approach is good, and is similar to the one I used for Remix. A good future addition could potentially be to automatically add the prefix for all node builtins when reolving the ID.

**Comment by alexanderniebuhr** [comment]
Looked at the testing, and I guess it's caused because we don't mark node built-ins as `external` (https://github.com/withastro/adapters/pull/286/files#diff-ee633f26523d8c890908778b42213c6fa4b86384998edff496759673f7b9ff95L312)

@ascorbic does Netlify support imports from `node_modules` folder, or do we also need to bundle all dependencies inline?

File: packages/netlify/test/functions/fixtures/middleware/src/middleware.ts [hunk 1]
```diff
@@ -1,5 +1,10 @@
+import http from 'http';
```
Comment (alexanderniebuhr) [inline packages/netlify/test/functions/fixtures/middleware/src/middleware.ts:1]:
```suggestion
```

File: packages/netlify/test/functions/fixtures/middleware/src/middleware.ts [hunk 2]
```diff
@@ -1,5 +1,10 @@
...
+import https from 'node:https';
+
 export const onRequest = (context, next) => {
 	context.locals.title = 'Middleware';
+	context.locals.nodeUnprefixedImportExists = !!http;
```
Comment (alexanderniebuhr) [inline packages/netlify/test/functions/fixtures/middleware/src/middleware.ts:6]:
```suggestion
```

File: .changeset/silly-snails-search.md [hunk 3]
```diff
@@ -0,0 +1,5 @@
+---
+'@astrojs/netlify': patch
+---
+
+allow `node:` prefixed imports
```
Comment (alexanderniebuhr) [inline .changeset/silly-snails-search.md:5]:
```suggestion
Allows support for `node:` prefixed imports if using  the Adapter with `edgeMiddleware: true`
```

**Review by alexanderniebuhr — APPROVED** [review]
LGTM

**Pull request merged**